python/main.py

```diff
@@ -1,45 +1,100 @@
-import os
-import logging
-import pathlib
-from fastapi import FastAPI, Form, HTTPException
-from fastapi.responses import FileResponse
-from fastapi.middleware.cors import CORSMiddleware
-
-app = FastAPI()
-logger = logging.getLogger("uvicorn")
-logger.level = logging.INFO
-images = pathlib.Path(__file__).parent.resolve() / "images"
-origins = [os.environ.get("FRONT_URL", "http://localhost:3000")]
-app.add_middleware(
-    CORSMiddleware,
-    allow_origins=origins,
-    allow_credentials=False,
-    allow_methods=["GET", "POST", "PUT", "DELETE"],
-    allow_headers=["*"],
-)
-
-
-@app.get("/")
-def root():
-    return {"message": "Hello, world!"}
-
-
-@app.post("/items")
-def add_item(name: str = Form(...)):
-    logger.info(f"Receive item: {name}")
-    return {"message": f"item received: {name}"}
-
-
-@app.get("/image/{image_name}")
-async def get_image(image_name):
-    # Create image path
-    image = images / image_name
-
-    if not image_name.endswith(".jpg"):
-        raise HTTPException(status_code=400, detail="Image path does not end with .jpg")
-
-    if not image.exists():
-        logger.debug(f"Image not found: {image}")
-        image = images / "default.jpg"
-
-    return FileResponse(image)
+import os
+import logging
+import pathlib
+import json
+import requests
+import hashlib
+from fastapi import FastAPI, Form, HTTPException,File, UploadFile
+from fastapi.responses import FileResponse
+from fastapi.middleware.cors import CORSMiddleware
+
+app = FastAPI()
+logger = logging.getLogger("uvicorn")
+# logger level INFO->DEBUG
+logger.level = logging.DEBUG
+images = pathlib.Path(__file__).parent.resolve() / "images"
+items_file = pathlib.Path(__file__).parent.resolve() / "items.json"
+origins = [os.environ.get("FRONT_URL", "http://localhost:3000")]
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=origins,
+    allow_credentials=False,
+    allow_methods=["GET", "POST", "PUT", "DELETE"],
+    allow_headers=["*"],
+)
+
+
+@app.get("/")
+def root():
+    return {"message": "Hello, world!"}
+
+
+@app.post("/items")
+def add_item(name: str = Form(...),category: str = Form(...),image: UploadFile = File(...)):
+    logger.info(f"Receive item: {name}")
+    logger.info(f"Receive item: {category}")
+    logger.info(f"Receive item: {image}")
+
+    image_name = image.filename
+    hashed_image = get_hash(image_name)
+
+    save_items_in_file(name,category,hashed_image) 
+    save_image(image,hashed_image) 
+
+    return {"message": f"item received: {name}"}
+
+def save_items_in_file(name,category,image_name):
+    new_item = {"name": name, "category": category, "image_name": image_name}
+    if os.path.exists(items_file):

```

Comment:
どっちでもいいんですが、他のpath関連がos.pathではなくpathlibを使っているので、揃えてpathlibの機能を使うとスマートですね
```suggestion
    if items_file.exists():
```
https://docs.python.org/ja/3/library/pathlib.html